index.bs

```diff
@@ -1817,26 +1817,32 @@ algorithm-hash pairs to be specified.
 It is important to note that Web PKI provides additional security
 mechanisms in addition to simply establishing a chain of trust for a server
 name.  One of them is handling certificate revocation.  In cases where the
-certificate used is ephemeral, such mechanism is not necessary.  In other
+certificate used is ephemeral, such a mechanism is not necessary.  In other
 cases, the Web application has to consider the mechanism by which the
 certificate hashes are provisioned; for instance, if the hash is provided as a
 cached HTTP resource, the cache needs to be invalidated if the corresponding
-certificate has been rotated due to compromise.
+certificate has been rotated due to compromise.  Another security feature
+provided by the Web PKI are safeguards against certain issues with key
+generation, such as rejecting certificates with well-known weak keys; while
+this specification does not provide any specific guidance, browsers MAY reject
+those as a part of implementation-defined behavior.
 
 Web PKI enforces an expiry period requirement on the certificates.  This
 requirement limits the scope of potential key compromise; it also forces server
 operators to design systems that support and actively perform key rotation.
 For this reason, WebTransport imposes a similar expiry requirement; as the
 certificates are expected to be ephemeral or short-lived, the expiry period is
-limited to two weeks.  This also improves the cryptographic agility of the
-system, as the keys in the Subject Public Key Info field can be upgraded to a
-different algorithm as a part of the key rotation process.  WebTransport API
-lets the application to specify multiple certificate hashes at once, allowing
-the client to accept multiple certificates for a perioid in which a new
-certificate is being rolled out.
+limited to two weeks.  The two weeks limit is a balance between setting the
+expiry limit as low as possible to minimize consequences of a key compromise,
+and maintaining it sufficiently high to lower the operational costs of
+provisioning certificates.
```

Comment:
This reasoning seems a little odd.  Part of the point of using hashes is so that you don't need to provision certificates in the usual fashion and this might read as saying that.

What I think you are talking about is the potential costs of synchronizing configuration across the resources that depend on the service (and so have certificate hashes embedded) and the systems that terminate connections.  Having to synchronize updates of these systems is made easier if the validity period is longer.

You also don't mention the problem of client clocks being horrible out of sync, which is relevant here.

Reply:
"Synchronization" is probably more appropriate here than "provisioning", indeed.  Reworded and added a mention of clock sync.